src/Symfony/Component/Security/Http/EventListener/IsCsrfTokenValidAttributeListener.php

```diff
@@ -50,7 +50,8 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                 continue;
             }
 
-            if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($id, $request->getPayload()->getString($attribute->tokenKey)))) {
+            $tokenValue = $this->getTokenValue($request, $attribute->tokenSource, $attribute->tokenKey);
+            if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($id, $tokenValue))) {
```

Comment:
this does not properly handle the case where `getTokenValue` returns `null`

Reply:
well, I checked: CsrfToken accepts null as $tokenValue, and it's turned into the empty string internally, so that then it's the responsibility of the manager to decide. Looks appropriate

Reply:
```suggestion
            if (
                null === $tokenValue 
                || !$this->csrfTokenManager->isTokenValid(new CsrfToken($id, $tokenValue))
            ) {
                throw new InvalidCsrfTokenException('Invalid CSRF token.');
            }
```

?  to avoid an unwanted call to isTokenValid that should be always false.

Reply:
:thinking:  But if I change as suggested

<img width="1829" height="1370" alt="image" src="https://github.com/user-attachments/assets/98590852-8efb-499a-81ce-3486b72dd52b" />

The existing test `testIsCsrfTokenValidCalledCorrectlyWithInvalidTokenKey` was previously valid, due to the fact that CsrfToken with a $value null/'' will return true. (A bit weird to me..)
https://github.com/symfony/symfony/blob/7.4/src/Symfony/Component/Security/Http/Tests/EventListener/IsCsrfTokenValidAttributeListenerTest.php#L171-L174

With my change, it will now throw an `Invalid CSRF token` exception. This raises the question of whether this is a BC break or a bug fix... (I tend to consider it a bug fix)

Reply:
it's just the test case that's bad